README.md

```diff
@@ -1,26 +1,255 @@
+sudo apt update
+
+# desktop-Full Install: (Recommended) : ROS, rqt, rviz, robot-generic libraries, 2D/3D simulators and 2D/3D perception
+sudo apt install ros-melodic-desktop-full
+```
+
+Dependencies for building packages
+```
+# install 'em
+sudo apt install python-rosdep python-rosinstall python-rosinstall-generator python-wstool build-essential
+
+# initialize rosdep
+sudo apt install python-rosdep
+sudo rosdep init
+rosdep update
+
+# sensor msgs dependencies
+rosinstall_generator sensor_msgs --rosdistro melodic --deps --wet-only --tar >melodic-sensor_msgs-wet.rosinstall
+wstool init src melodic-sensor_msgs-wet.rosinstall
+```
+
+#### 2.2.2 ROS Kinetic on Raspbian
+
+Use these instructions if you followed subsection 2.1.2 above.
+
+If you do not download the pre-built Raspbian Stretch / ROS image, you will need to build ROS yourself.
+
+First, make sure you update all the packages on your pi:
+
+```
+sudo apt-get update
+sudo apt-get upgrade
+```
+
+(The above commands may take a few minutes to run.)
+
+Next, we will install ROS (and specifically, the version called ’Kinetic’).
+
+**NOTE: Follow the ROS installation directions in the link below CAREFULLY! Do not ignore warnings, as they will lead to issues later in the installation. Some known ”gotchas” to note before you get started are:**
+
+- Make sure you install the Kinetic ”desktop full” version of ROS, not the desktop ros-comm, or other versions.
+- ROS changed their GPG keys as of June 7 2019. The ROS install instructions do not mention this as of July 2019 and will not work unless you update your GPG keys. See [here](https://discourse.ros.org/t/new-gpg-keys-deployed-for-packages-ros-org/9454) for instructions on how to update your GPG keys.
+- Make sure you follow directions for KINETIC, not Indigo, Melodic, or any other ROS version.
+
+Once you familiarize yourself with the above issues, go ahead and complete your ROS installation by following these [instructions](http://wiki.ros.org/ROSberryPi/Installing%20ROS%20Kinetic%20on%20the%20Raspberry%20Pi)
+
+
+### 2.3 Setting up serial communication on the RPi
+
+#### 2.3.1 Ubuntu 18.04
+
+You'll need to create dev files for serial0 and serial1 and update permissions:
+```
+sudo ln -s /dev/ttyS0 /dev/serial0
+sudo ln -s /dev/ttyAMA0 /dev/serial1
+
+# change the group of the new serial devices
+sudo chgrp -h tty /dev/serial0
+sudo chgrp -h tty /dev/serial1
 
-1. Follow the instructions to set up your Raspberry Pi and install ROS in 
-[Software Steps](https://github.com/nasa-jpl/open-source-rover/blob/master/Software/Software%20Steps.pdf)
-2. In the newly created catkin workspace you just made, clone this repo:
+# The devices are now under the tty group. Need to add the ubuntu user to the tty group:
+sudo adduser ubuntu tty
+# also need dialout for ttyS0 and ttyAMA0
+sudo adduser ubuntu dialout
+
+# update the permissions for group read on the devices
+sudo chmod g+r /dev/ttyS0
+sudo chmod g+r /dev/ttyAMA0
```

Comment:
All of this happens through the udev file. I'm not sure if these steps are required for RPi3

Reply:
I never fully vetted that tbh. I need to follow this README from scratch with a fresh SD and fully check everything, methinks.

Reply:
@Achllle you mean this stuff https://www.tapatalk.com/groups/jpl_opensource_rover/viewtopic.php?p=1527#p1527 ?

Where does the group get set for `serial0` and `serial1`? I actually forget how they ended up as `tty` for me, but seems appropriate?

Reply:
This  is fixed in the latest. Used my own homebrewed udev rules file, so that only the bare minimum would be in there.